File: Algorithms/Sort Algorithms/Insertion Sort/insert.js
const insert=(array)=>{
    
    for(let i = 1;i < array.length;i++){
        let temp = array[i]
        let j = i - 1

        while(j >= 0 && array[j] > temp){
            array[j+1] = array[j]
            j--
        }
        array[j+1] = temp
    }
    return array
}


let testArray = [8, 5, 3, 2, 7, 1]
let sortedArray = insert(testArray)
console.log(sortedArray)

File: Algorithms/Sort Algorithms/Insertion Sort/insert.py
def insert(array):
    for i in range(1,len(array)):
        temp = array[i]
        j = i-1

        while j>=0 and array[j] > temp:
            array[j+1] = array[j]
            j -= 1

        array[j+1] = temp

    return array

test_array = [8,4,6,3,2,5,1]
sorted_array = insert(test_array)
print(sorted_array)        